Lib/_collections_abc.py

```diff
@@ -409,6 +413,80 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _CallableGenericAlias(GenericAlias):
+    """ Represent `Callable[argtypes, resulttype]`.
+
+    This sets ``__args__`` to a tuple containing the flattened``argtypes``
+    followed by ``resulttype``.
+
+    Example: ``Callable[[int, str], float]`` sets ``__args__`` to
+    ``(int, str, float)``.
+    """
+
+    __slots__ = ()
+
+    def __new__(cls, origin, args):
+        try:
+            return cls.__create_ga(origin, args)
+        except TypeError as exc:
+            if sys.version_info[:2] == (3, 9):
+                import warnings
+                warnings.warn(f'{str(exc)} '
+                              f'(This will raise TypeError in Python 3.10)',
+                              DeprecationWarning)
+                return GenericAlias(origin, args)
+            else:
+                raise exc
```

Comment:
This should probably be just `raise` (not `raise exc`). But really, I think we just need to check in different code in the 3.9 branch (and we can refine the 3.9 code after the 3.10 code is landed). Sorry about the misunderstanding.

Reply:
Ok, and no worries, I misinterpreted that. Btw for the sake of easier backporting any future changes (I foresee ParamSpec being one of them), I'm going to separate out the method call.

Reply:
Woops I just remembered ParamSpec is 3.10, I meant any bugfixes to Callable 😉 .